### PR TITLE
Fix / Rate oracle conversions on XEMM strategy

### PR DESCRIPTION
**Pull request by TheHolyRoger**
**Before submitting this PR, please make sure**:

- [X] Your code builds clean without any errors or warnings
- [X] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
@Nullably @RC-13 
I found some issues with rate conversion that are fixed in this PR, can you confirm the issues/fixes and that I've not made an error myself?

### CODE REVIEW

File: hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx [hunk 1]
```diff
@@ -1135,13 +1135,16 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
...
             ExchangeBase taker_market = market_pair.taker.market
-
-            object quote_asset_amount = maker_market.c_get_balance(market_pair.maker.quote_asset) if is_buy else \
-                taker_market.c_get_balance(market_pair.taker.quote_asset)
-            object base_asset_amount = taker_market.c_get_balance(market_pair.taker.base_asset) if is_buy else \
-                maker_market.c_get_balance(market_pair.maker.base_asset)
             object order_size_limit
 
+        quote_pair, quote_rate_source, quote_rate, base_pair, base_rate_source, base_rate = \
+            self.get_taker_to_maker_conversion_rate()
```
Comment (TheHolyRoger) [inline hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx:1141]: Taker Quote and Base balances were not rate converted.

File: hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx [hunk 2]
```diff
@@ -1161,11 +1164,8 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
         """
         Return price conversion rate for a taker market (to convert it into maker base asset value)
         """
-        if not self._use_oracle_conversion_rate:
-            return self._taker_to_maker_quote_conversion_rate / self._taker_to_maker_base_conversion_rate
-        else:
-            _, _, quote_rate, _, _, base_rate = self.get_taker_to_maker_conversion_rate()
-            return quote_rate / base_rate
+        _, _, quote_rate, _, _, base_rate = self.get_taker_to_maker_conversion_rate()
```
Comment (TheHolyRoger) [inline hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx:1167]: Redundant check here with the new function

File: hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx [hunk 3]
```diff
@@ -1202,15 +1202,15 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
...
                     if self._logging_options & self.OPTION_LOG_CREATE_ORDER:
                         self.log_with_clock(
                             logging.INFO,
                             f"({market_pair.maker.trading_pair}) Creating limit bid order for "
                             f"{bid_size} {market_pair.maker.base_asset} at "
                             f"{bid_price} {market_pair.maker.quote_asset}. "
-                            f"Current hedging price: {effective_hedging_price} {market_pair.taker.quote_asset} "
-                            f"(Rate adjusted: {effective_hedging_price_adjusted:.2f} {market_pair.taker.quote_asset})."
+                            f"Current hedging price: {effective_hedging_price:.8f} {market_pair.maker.quote_asset} "
+                            f"(Rate adjusted: {effective_hedging_price_adjusted:.8f} {market_pair.taker.quote_asset})."
```
Comment (TheHolyRoger) [inline hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx:1213]: This log output for hedging price appeared to be completely wrong for me until these changes...?

File: hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx [hunk 4]
```diff
@@ -1241,15 +1241,15 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
                         False,
                         ask_size
                     )
-                    effective_hedging_price_adjusted = effective_hedging_price
+                    effective_hedging_price_adjusted = effective_hedging_price / self.market_conversion_rate()
```
Comment (TheHolyRoger) [inline hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx:1244]: Hedge price in log output was not divided by conversion rate

File: hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx [hunk 5]
```diff
@@ -1241,15 +1241,15 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
...
                     if self._logging_options & self.OPTION_LOG_CREATE_ORDER:
                         self.log_with_clock(
                             logging.INFO,
                             f"({market_pair.maker.trading_pair}) Creating limit ask order for "
                             f"{ask_size} {market_pair.maker.base_asset} at "
                             f"{ask_price} {market_pair.maker.quote_asset}. "
-                            f"Current hedging price: {effective_hedging_price} {market_pair.maker.quote_asset} "
-                            f"(Rate adjusted: {effective_hedging_price_adjusted:.2f} {market_pair.maker.quote_asset})."
+                            f"Current hedging price: {effective_hedging_price:.8f} {market_pair.maker.quote_asset} "
+                            f"(Rate adjusted: {effective_hedging_price_adjusted:.8f} {market_pair.taker.quote_asset})."
```
Comment (TheHolyRoger) [inline hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx:1252]: When working with some tokens, 2 decimal places isn't enough for rate converted price

**Review by Nullably — APPROVED** [review]
@TheHolyRoger Thanks a lot for catching all these issues. Some, perhaps most, of these are existing issues that we were not aware of.

**Pull request merged**